docs/tutorials/infrastructure/kne/srl-with-kne/installation.md

```diff
@@ -0,0 +1,119 @@
+
+```bash
+kne deploy deploy/kne/kind-bridge.yaml # (1)!
+```
+
+1. run from the root of the kne repository. Also remove the
+
+The deployment process should finish with the `Deployment complete, ready for topology` log message, indicating that every component of a KNE cluster has been successfully deployed. At this point, it is helpful to check that the cluster and its components are running healthy.
+
+=== "kind cluster"
+    Ensure that a kind cluster named `kne` is active.
+
+    ```bash
+    ❯ kind get clusters
+    kne
+    ```
+
+    Check that `kubectl` is configured to work with `kne` cluster:
+
+    ```bash
+    ❯ kubectl config current-context
+    kind-kne
+    ```
+=== "CNI"
+    Ensure that `meshnet` CNI is running as a daemonset:
+
+    ```bash
+    ❯ kubectl get daemonset -n meshnet
+    NAME      DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR              AGE
+    meshnet   1         1         1       1            1           kubernetes.io/arch=amd64   8m55s
+    ```
+
+=== "Load Balancer"
+    Ensure that MetalLB Load Balancer is running controller deployment and speaker daemonset:
+
+    ```bash
+    ❯ kubectl get pod -n metallb-system
+    NAME                          READY   STATUS    RESTARTS   AGE
+    controller-55d86f5f7c-bl9kx   1/1     Running   0          12m
+    speaker-zsj29                 1/1     Running   0          11m
+    ```
+
+## SR Linux controller
+
+Next step is to install [SR Linux controller](https://github.com/srl-labs/srl-controller) that provides the necessary APIs for KNE to deploy SR Linux nodes as part of the network topology. SR Linux controller is an open-source project hosted at :material-github: [srl-labs/srl-controller](https://github.com/srl-labs/srl-controller) repository and can be easily installed on a k8s cluster as per its [installation instructions](https://github.com/srl-labs/srl-controller#install):
+
+```bash
+kubectl apply -k https://github.com/srl-labs/srl-controller/config/default
+```
+
+!!!info
+    Additional controllers can be installed by following the respective installation instructions provided in the [KNE documentation](https://github.com/openconfig/kne/blob/main/docs/create_topology.md#deploying-additional-vendor-controllers).
+
+When `srl-controller` is installed successfully, it can be seen in its namespace as a deployment:
+
+```bash
+❯ kubectl get deployments -n srlinux-controller
+NAME                                    READY   UP-TO-DATE   AVAILABLE   AGE
+srlinux-controller-controller-manager   1/1     1            1           12m
+```
+
+### License
+
+If a user intends to run a topology with chassis-based SR Linux nodes[^1] they need to [install a valid license](https://github.com/srl-labs/srl-controller/blob/main/docs/using-licenses.md).
```

Comment:
are the chassis-based nodes required for the OC services? this adds an extra non-automated step to the setup so if possible I would like to avoid this

but if using those services requires the chassis-based version then I will need to figure out a way to automatically setup the license (for our automated tests)

Reply:
also is there a way to add an error for missing license to the controller or pods? right now it just hangs

Reply:
also how do we get a license? should the link here point to a doc that also includes info about how to create the license file (who do we contact for a license?)

Reply:
At this moment OC services are only available on the chassis based systems that require a license. 
I am confident that Ryan Elliott is the mediator on your side that receives licenses from us. They are generated on our end. 

Once you have a license you can either  upload it to the cluster as explained in the docs, or to embed it to an srlinux image so that you don't need to touch the automatically provisioned cluster. 

Because of that second option that some people might choose to use I'm reluctant on adding a hard stop when the license config map is absent in the cluster. 

Wdyt, will embedding a license into the image be the choice for ondatra given that you are not using images from upstream repositories directly?

Reply:
Responding over email